File: euler/problem4.js
/*
A palindromic number reads the same both ways. The largest palindrome made from 
the product of two 2-digit numbers is 9009 = 91 × 99.

Find the largest palindrome made from the product of two 3-digit numbers.

*/

var palindromicNumber = function(n){
  var upperBound = Math.pow(10, n) - 1;
  var answer;
  return upperBound;
};

console.log(palindromicNumber(3));

function numIsPal(number){
  var numStr = number.toString();
  var i = 0;
  var j = numStr.length - 1;
  var flag = true;
  while(i <= j && flag){
    if(numStr[i] !== numStr[j]){
      flag = false;
    }
    i++;
    j--;
  }
  return flag;
}

File: euler/problem4.py
def isPalindrome(num):
  numStr = str(num)
  i = 0
  j = len(numStr) - 1
  while(i <= j):
    if(numStr[i] != numStr[j]):
      return False
    i+= 1
    j-= 1
  return True

def largestPalindrome(digits):
  largestPal = 0
  upperBound =  10 ** digits - 1
  for i in range(upperBound - (10 ** (digits - 1)), upperBound):
    for j in range(upperBound - (10 ** (digits - 1)), upperBound):
      product = i * j
      if isPalindrome(product) and product > largestPal:
        largestPal = product
  return largestPal


print largestPalindrome(3)